### DISCUSSION TITLE
GCP Support

### DISCUSSION THREAD

**Discussion by ChakshuGautam** [Sunbird-Ed/Community#224] [Installation] [answered]
Hi Maintainers, 

Sharing more details about the issue we are facing on the Sunbird GCP deployment front.

**Context**: UP Human Resources in Health (Niramaya) is a program aimed at improving the quality of training imparted to nursing and paramedical human resources, thereby bettering the quality of health services across the state. We are seeking to target 50,000+ students across 800+ institutes. There are five key reforms being driven through this project:
1) Public accreditation of institutes imparting trainings to paramedics & nurses to increase compliance with norms and aid students in selecting institutions during application process
2) Centralized admissions for all nursing and paramedical courses across the state
3) Competency based assessments based on learning objective for students and professionals
4) Verifiable e credentials to students to establish credibility when seeking employment 
5) Career awareness & employment linkages, thereby making health care jobs more lucrative for students

Sunbird building blocks are being leveraged to to drive this program & Tarento has been onboarded by the government for technical delivery. The additional development done will be contributed back to the Sunbird repository.

**Issue**: Our project needs deployment of all sunbird building blocks on GCP, however, when it comes to file storage, all modules are S3 compliant & not GCP compliant. These gaps were identified when Diksha was deployed on GCP and are being looked into by the Sunbird team.

**Our involvement**:  I have been in touch with the Sunbird developers on Slack to understand the progress being made. However, we still don't have clarity into what is left and the ETA for completion. I got hold of a [workplan]( https://docs.google.com/spreadsheets/d/1cqIksAfKF_5lHlcl0xoi0bdN0JOpeOr1gs2UgIEcR7s/edit?usp=drivesdk) (this is a copy), but as we understand this was last updated in September.

**Ask**: Would be great if we can create a public ticket on this front so that everyone is aware of the progress happening on this front.

Let us know if you have any queries. Look forward to your support.

**Comment by ChakshuGautam** [comment, marked as answer]
@surendrasinghs @santhosh-tg @sharathprasad4tarento @rahul101001000

**Reply by sharathkashyap** [reply]
@ChakshuGautam Most of the Sunbird BB teams are currently working towards support for Hyper Scaler CSPs. SunbirdED Plans to adopt these changes during release of 5.2.0 (Mid January). However, the current SunbirdED supports the Infra to be hosted on major CSP players. There is a hard wired dependency on Object Storage and Media Streaming Services. 
If in case project is asking for GCP support. All the cloud related changes might be checked in as part of 5.1.0.

Team may have to start writing extensions on following modules:
a) Ensure the streaming media URLs generated are specific to CSP Provider. So that the Bulk of Bandwidth usage happens from GCP.(**Changes in Knowlg**)
b) All newly created and Published content to be modified to GCP urls for assets, thumbnails, ecar urls. (**Some changes in Knowlg and InQuiry**)
c) Some Plugin and Library related dependencies : SunbirdED is currently working along with other BBs to move the asset URLs generated to CNAME based URLs. Suggest you to wait for these changes alone from us.

**Reply by chitranshu-keshav** [reply]
@ChakshuGautam Can you please mark this thread as answered. If your issue is not resolved then please follow up with the relevant information
